### PR TITLE
CFE-2918: Fixed small memory leaks of environment variable strings

### PR DESCRIPTION
**Pull request by olehermanse**
Replaces #3341

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thank you for submitting a PR! Maybe @vpodzime can review this?

**Comment by olehermanse** [comment]
@cf-bottom jenkins with exotics, please.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1371/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/1371/

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1376/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/1376/

**Comment by olehermanse** [comment]
@cf-bottom jenkins with exotics, please.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1394/)

(with exotics)

https://ci.cfengine.com/job/pr-pipeline/1394/

**Comment by olehermanse** [comment]
@vpodzime This is passing now. Please review.

File: libutils/misc_lib.c [hunk 1]
```diff
@@ -26,7 +26,9 @@
 
 #include <platform.h>
 #include <alloc.h>
+#include <string_lib.h>
 #include <logging.h>
+#include <sequence.h>
```
Comment (vpodzime) [inline libutils/misc_lib.c]: I think this no longer needs `sequence.h`

Comment (vpodzime) [reply libutils/misc_lib.c]: But I think you should `#include <errno.h>`

Comment (olehermanse) [reply libutils/misc_lib.c]: Will make a follow up.

**Pull request merged**